src/adaptation/stepsize.jl

```diff
@@ -17,22 +17,34 @@ end
 
 function DAState(ϵ::AbstractVector{T}) where {T}
     n = length(ϵ)
-    μ = computeμ(ϵ)
+    μ = map(computeμ, ϵ)
     return DAState(0, ϵ, μ, zeros(T, n), zeros(T, n))
 end
 
 function reset!(das::DAState{T}) where {T<:AbstractFloat}
     das.m = 0
     das.μ = computeμ(das.ϵ)
     das.x_bar = zero(T)
-    return das.H_bar = zero(T)
+    das.H_bar = zero(T)
+    return das
 end
 
 function reset!(das::DAState{<:AbstractVector{T}}) where {T<:AbstractFloat}
     das.m = 0
-    das.μ .= computeμ(das.ϵ)
-    das.x_bar .= zero(T)
-    return das.H_bar .= zero(T)
+    map!(computeμ, das.μ, das.ϵ)
```

Comment:
Isn't the `.=` statement storing results in LHS in place too? Can you explain the difference between `.=` and `map!`? Sorry, I haven't used `map!` or `fill!` extensively before.